ethers-core/src/types/transaction/eip2718.rs

```diff
@@ -909,4 +909,12 @@ mod tests {
             assert_eq!(tx0, tx1);
         }
     }
+
+    #[test]
+    fn malformed_transaction_does_not_panic() {
+        let tx = hex::decode("b9011e02f9011a0180850ba43b7400850df8475864830493e0947a250d5630b4cf539739df2c5dacb4c659f2488d88016345785d8a0000b8e4b6f9de95000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000800000000000000000000000005bd929b84c3dac16515c6b6c95f70fdae9c05c0c00000000000000000000000000000000000000000000000000000000650267140000000000000000000000000000000000000000000000000000000000000002000000000000000000000000c02aaa39b223fe8d0a0e5c4f27ead9083c756cc2000000000000000000000000dac17f958d2ee523a2206206994597c13d831ec7c0808080").unwrap();
```

Comment:
where's this coming from?

Reply:
This was given to me - apparently it was spit out from go-ethereum. But I also have encountered the same issue while fuzzing inputs to this function. Regardless of the exact payload, I don't think think we want it to panic. If you run the test on the [commit prior ]([9ad0740](https://github.com/gakonst/ethers-rs/pull/2694/commits/9ad074096f83c1f70251c1a1eadf8c3c281f1150))to the fix being introduced, it will panic.